napari/layers/base/base.py

```diff
@@ -1304,38 +1226,41 @@ def _world_to_displayed_data_ray(
         vector_data_ndisplay /= np.linalg.norm(vector_data_ndisplay)
         return vector_data_ndisplay
 
-    def _world_to_data_dims_displayed(
-        self, dims_displayed: List[int], ndim_world: int
+    def _world_to_layer_dims(
+        self, *, world_dims: List[int], ndim_world: int
     ) -> List[int]:
-        """Convert indices of displayed dims from world to data coordinates.
+        """Map world dimensions to layer dimensions while maintaining order.
 
-        This accounts for differences in dimensionality between the world
-        and the data coordinates. For example a world dims order of
-        [2, 1, 0, 3] would be [0, 1] for a layer that only has two dimensions
-        or [1, 0, 2] for a layer with three as that corresponds to the
-        relative order of the last two and three dimensions respectively
+        This is used to map dimensions from the full world space defined by ``Dims``
+        to the subspace that a layer inhabits, so that those can be used to index the
+        layer's data and associated coordinates.
+
+        For example a world ``Dims.order`` of [2, 1, 0, 3] would map to [0, 1] for a
+        layer with two dimensions and [1, 0, 2] for a layer with three dimensions
+        as those correspond to the relative order of the last two and three world dimensions
+        respectively.
 
         Parameters
         ----------
-        dims_displayed : List[int]
-            The world displayed dimensions.
+        world_dims : List[int]
+            The world dimensions.
         ndim_world : int
             The number of dimensions in the world coordinate system.
 
         Returns
         -------
-        dims_displayed_data : List[int]
-            The displayed dimensions in data coordinates.
+        List[int]
+            The corresponding layer dimensions with the same ordering as the given world dimensions.
         """
         offset = ndim_world - self.ndim
-        order = np.array(dims_displayed)
+        order = np.array(world_dims)
         if offset <= 0:
```

Comment:
I think we expect `offset` to always be non-negative (because the number of dimensions in the world should always be greater than equal to the number of dimensions in any one layer), so I wonder if we could just remove this condition. But more informed opinions are welcome!

Reply:
I agree with your expectation here and don't know why this would ever be negative, possibly some strange initialisation?

Reply:
Could this come into play when a new layer with more dims than thew world gets added to the viewer?

Reply:
My assumption is that is intended to never happen (because `Dims` should be updated first) and the order of the code right now in `ViewerModel._on_add_layer` means that it doesn't. I guess it's possible for a layer that hasn't been added yet to the viewer to trigger this function, though I'm unsure of any public ways to do that. Again, I assume the intent is that calling this only makes sense for layers that have been added to the viewer given that it is finding correspondences between the layer's dimensions and those displayed (presumably in the viewer).